app/src/main/java/com/base/firebase_operations/MainActivity3.kt

```diff
@@ -0,0 +1,49 @@
+package com.base.firebase_operations
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.ktx.storage
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class MainActivity3 : AppCompatActivity() {
+    lateinit var recyclerview:RecyclerView
+    val imagereff=Firebase.storage.reference
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main3)
+        recyclerview=findViewById(R.id.recycler1)
+        getfilefromCloud()
+    }
+
+    private fun getfilefromCloud()= CoroutineScope(Dispatchers.IO).launch {
```

Comment:
This function is basically used to get the all the images that are present in the Firebase cloud storage,First we create the reference variable which would be pointing towards the images location stored in the database,after that we use the listALL() function to get a list of all items that are the images located in the directory passed.
Then we just created a mutable list which is used to store URL of the downloaded images and the reason for storing the url is that we would use the Glide library for loading those urls into the image view:
Now we iterate over each item present in the imagesfromcloud like all the images and then we would first download the url of each image then once it is downloaded we would add it to our mutable list.
Now we just changed our Coroutine Context for accessing the UI elements like in this case we are displaying the images inside the recycler view and just create the adapter variable which would be pointing to the adapter class which we created for our recycler view and pass the list that we obtained from the above execution and set that adapter to our recycler view.